File: Assets/Typescript~/src/Shared/Queue/QueueDefinitions.ts
import { Theme } from "@Easy/Core/Shared/Util/Theme";
import { QueueMeta } from "./QueueMeta";
import { QueueType } from "./QueueType";

export const Queues: Record<QueueType, QueueMeta> = {
	[QueueType.CLASSIC_SQUADS]: {
		name: "Classic (Squads)",
		teams: [
			{ id: "1", name: "Blue", maxPlayers: 4, color: Theme.teamColor.Blue },
			{ id: "2", name: "Red", maxPlayers: 4, color: Theme.teamColor.Red },
			{ id: "3", name: "Green", maxPlayers: 4, color: Theme.teamColor.Green },
			{ id: "4", name: "Yellow", maxPlayers: 4, color: Theme.teamColor.Yellow },
		],
		maps: ["Aztec"],
	},
};

File: Assets/Bundles/Shared/Resources/TS/Queue/QueueDefinitions.lua
-- Compiled with unity-ts v2.1.0-75
local Theme = require("Shared/TS/Util/Theme").Theme
local QueueType = require("Shared/TS/Queue/QueueType").QueueType
local GameMap = require("Server/TS/Services/Match/Map/Maps").GameMap
local Queues = {
	[QueueType.CLASSIC_SQUADS] = {
		name = "Classic (Squads)",
		teams = { {
			id = "1",
			name = "Blue",
			maxPlayers = 4,
			color = Theme.Blue,
		}, {
			id = "2",
			name = "Red",
			maxPlayers = 4,
			color = Theme.Red,
		}, {
			id = "3",
			name = "Green",
			maxPlayers = 4,
			color = Theme.Green,
		}, {
			id = "4",
			name = "Yellow",
			maxPlayers = 4,
			color = Theme.Yellow,
		} },
		maps = { GameMap.SANCTUM },
	},
}
return {
	Queues = Queues,
}
-- ----------------------------------
-- ----------------------------------
